internal/observability/call.go

```diff
@@ -496,6 +499,17 @@ func (c call) logStreamEvent(err error, success bool, succMsg, errMsg string, ex
 	ce.Write(fields...)
 }
 
+// emitSpanErrorTag sets the error information as tags on the current span
+func (c call) emitSpanErrorTag(res callResult) {
+	if span := opentracing.SpanFromContext(c.ctx); span != nil {
+		// emit application error to span tag if applicable
+		if res.isApplicationError {
```

Comment:
Let's check this boolean first (if !.. {return}), it's much cheaper than creating span from context.